.github/workflows/sonarcloud.yml

```diff
@@ -4,33 +4,58 @@ on:
   push:
     branches: [ "*" ]
   pull_request:
-    branches: [ "*" ]
+    types: [opened, synchronize, reopened]
   workflow_dispatch:
 
+env:
+  java-version: 17
+
 permissions:
-  pull-requests: read
+  pull-requests: write
 
 jobs:
   Analysis:
     runs-on: ubuntu-latest
 
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
 
-      - name: Set SonarCloud Project Key
-        run: |
-          REPO_NAME=$(echo $GITHUB_REPOSITORY | cut -d '/' -f 2)
-          ORG_NAME=$(echo $GITHUB_REPOSITORY | cut -d '/' -f 1)
-          SONAR_PROJECT_KEY="${ORG_NAME}_${REPO_NAME}"
-          echo "SONAR_PROJECT_KEY=$SONAR_PROJECT_KEY" >> $GITHUB_ENV
+      - name: AI Code Reviewer
+        uses: freeedcom/ai-codereviewer@main
```

Comment:
Please ensure that the repository name is correct. The original repository name was 'ai-code-reviewer' and you have changed it to 'ai-codereviewer'. If this is not a valid repository, this change will cause the workflow to fail.